# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Parent  0f2218a7f7d580ca5e81bb34fa37e7baef2dae4f
Fix issue when making messages unread manually.

diff --git a/mailnews/base/src/MailNotificationService.jsm b/mailnews/base/src/MailNotificationService.jsm
--- a/mailnews/base/src/MailNotificationService.jsm
+++ b/mailnews/base/src/MailNotificationService.jsm
@@ -223,20 +223,32 @@ NewMailNotificationService.prototype = {
           Math.max(folderCount + newCount, 0)
         );
       }
       /* eslint-enable no-lonely-if */
     }
 
     let folderNew = this.folderNewMap.get(folder.URI);
     if (!folderNew) {
-      this.folderNewMap.set(folder.URI, {
-        name: folder.name,
-        isInbox: folder.flags & Ci.nsMsgFolderFlags.Inbox,
-      });
+      // Reinstate 91/102 code.
+      this._log.debug(
+        `_updateNewCount: ${folder.URI} not in the map, returning`
+      );
+      return;
+      // The code below causes this error:
+      // If there are no new messages, the folder is not in the map.
+      // Continuing here will lose the temporary negative count
+      // and lead to a phantom new message which can't be cleared.
+      // Sadly we can't remember why the code was added in:
+      // https://github.com/Betterbird/thunderbird-patches/commit/8bfcd49bde38633e5fba11aff9cdceb143f1e77a
+      // and which issue it tried to address.
+      // this.folderNewMap.set(folder.URI, {
+      //  name: folder.name,
+      //  isInbox: folder.flags & Ci.nsMsgFolderFlags.Inbox,
+      //});
     }
 
     this._mNewCount = Math.max(this._mNewCount + newCount, 0);
     this._log.debug(
       `_updateNewCount: ${folder.URI} new mail count: ${this._mNewCount}`
     );
     if (this.countNew) {
       this._notifyListeners(
